client/app/components/dashboards/AddWidgetDialog.jsx

```diff
@@ -143,7 +140,8 @@ class AddWidgetDialog extends React.Component {
       .save()
       .then(() => {
         dashboard.widgets.push(widget);
-        this.props.close();
+        this.props.onAdded();
+        this.close();
```

Comment:
Will this also trigger `props.onClose`?

Reply:
Yep!
https://github.com/getredash/redash/blob/009cde7c347f0a5a1010c7af9ec22723c3487479/client/app/components/dashboards/AddWidgetDialog.jsx#L297

This allows the closing animation to happen, and only then `ng-if` removes the modal from the dom.